java/sage/io/LocalSageFile.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2015 The SageTV Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sage.io;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+
+public class LocalSageFile extends RandomAccessFile implements SageFileSource
```

Comment:
I'm curious why you chose to extend RandomAccessFile rather than make it a member inside of here? IIRC, the reason I didn't extend it was because I didn't want to end up using methods that I didn't expect to be used. I notice in yours you make some methods not supported, and so generally you're changing this to really not be like java.io.RandomAccessFile anymore (which is fine); but in that case you probably shouldn't make it extend it.

This class is also more about general byte access into a local file...and RandomAccessFile exposes all those other DataInput/DataOutput methods that it also seems to go beyond the scope of what LocalSageFile was intended to cover.